test/functional/tests/stress/test_kedr.py

```diff
@@ -0,0 +1,210 @@
+#
+
+import pytest
+
+from datetime import timedelta
+
+from test_tools.kedr import Kedr, KedrProfile
+from api.cas import cas_module, installer, casadm
+from core.test_run import TestRun
+from test_utils import os_utils
+from test_utils.size import Size, Unit
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+
+
+@pytest.fixture(scope="module")
+def install_kedr():
+    TestRun.LOGGER.info("Checking if kedr is installed")
+    if not Kedr.is_installed():
+        TestRun.LOGGER.info("Installing kedr")
+        Kedr.install()
+
+
+@pytest.fixture(scope="function")
+def unload_modules():
+    TestRun.LOGGER.info("Check if CAS is installed")
+    if installer.check_if_installed():
+        TestRun.LOGGER.info("Unloading modules")
+        cas_module.unload_all_cas_modules()
+
+    TestRun.LOGGER.info("Stop kedr if it is running")
+    if Kedr.is_loaded():
+        Kedr.stop()
+
+    TestRun.LOGGER.info("Mounting debugfs")
+    os_utils.mount_debugfs()
+
+
+@pytest.mark.parametrize("module", cas_module.CasModule)
+def test_kedr_memleak_load_cas_module(module, unload_modules, install_kedr):
+    """
+    title: Loading modules with kedr started with 'memleak' configuration
+    description: Load and unload modules with kedr started to watch for memmory leaks
+    pass_criteria:
+      - No memory leaks observed after loading and unloading module
+    """
+    with TestRun.step(f"Starting kedr against {module}"):
+        Kedr.start(module.value)
+
+    with TestRun.step(f"Loading {module}"):
+        os_utils.load_kernel_module(module.value)
+
+    with TestRun.step(f"Unloading {module}"):
+        os_utils.unload_kernel_module(module.value, os_utils.ModuleRemoveMethod.modprobe)
+
+    with TestRun.step(f"Checking for memory leaks for {module}"):
+        try:
+            Kedr.check_for_mem_leaks(module.value)
+        except Exception as e:
+            TestRun.fail(f"{e}")
+
+    with TestRun.step(f"Stopping kedr"):
+        Kedr.stop()
+
+
+@pytest.mark.parametrize("module", cas_module.CasModule)
+def test_kedr_fsim_load_cas_module(module, unload_modules, install_kedr):
+    """
+    title: Loading modules with kedr started with 'fsim' configuration
+    description: Load and unload modules with kedr started to simulate kmalloc fails
+    pass_criteria:
+      - Module fails to load
+    """
+    with TestRun.step(f"Starting kedr against {module}"):
+        Kedr.start(module.value, KedrProfile.FAULT_SIM)
+
+    with TestRun.step("Setting up fault simulation parameters"):
+        Kedr.setup_fault_injections()
+
+    with TestRun.step(f"Trying to load {module}"):
+        out = os_utils.load_kernel_module(module.value)
+        if out.exit_code == 0 \
+                or "Cannot allocate memory" not in out.stderr:
+            TestRun.fail(f"Loading module should fail because of alloc error, instead "
```

Comment:
Wouldn't it be better to not break the test here and allow stopping kedr in the next step? (in all tests)